ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,239 @@
+			return Task.CompletedTask;
+		}
+
+		MonitoringConfig? config = null;
+
+		foreach ((string configProperty, JsonElement configValue) in additionalConfigProperties) {
+			try {
+				config = configProperty switch {
+					nameof(Monitoring) => configValue.Deserialize<MonitoringConfig>(),
+					_ => config
+				};
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericException(e);
+				ASF.ArchiLogger.LogGenericWarning(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+				return Task.CompletedTask;
+			}
+		}
+
+		config ??= new MonitoringConfig();
+
+		if (!Enabled) {
+			ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+			return Task.CompletedTask;
+		}
+
+		Config = config;
+
+		ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginInitializedAndEnabled, nameof(MonitoringPlugin)));
+
+		return Task.CompletedTask;
+	}
+
+	public override Task OnLoaded() {
+		Utilities.WarnAboutIncompleteTranslation(Strings.ResourceManager);
+
+		Meter = new Meter(MeterName, Version.ToString());
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_memory_usage",
+			static () => GC.GetTotalMemory(false) / 1024,
+			description: "Current memory usage of ASF"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_ipc_banned_ips",
+			static () => ApiAuthenticationMiddleware.GetCurrentlyBannedIPs().Count(),
+			description: "Number of IP addresses currently banned by ASFs IPC module"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_active_plugins",
+			static () => PluginsCore.ActivePlugins.Count,
+			description: "Number of plugins currently loaded in ASF"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_bots", static () => {
+				ICollection<Bot> bots = Bot.Bots?.Values ?? ImmutableHashSet<Bot>.Empty;
+
+				return new List<Measurement<int>>(5) {
+					new(bots.Count, new KeyValuePair<string, object?>(TagNames.BotState, "configured")),
+					new(bots.Where(static bot => bot.IsConnectedAndLoggedOn).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "online")),
+					new(bots.Where(static bot => !bot.IsConnectedAndLoggedOn).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "offline")),
+					new(bots.Where(static bot => bot.CardsFarmer.NowFarming).Count(), new KeyValuePair<string, object?>(TagNames.BotState, "farming"))
+				};
+			},
+			description: "Number of bots that are currently loaded in ASF"
+		);
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_bot_friends", static () => {
+				ICollection<Bot> bots = Bot.Bots?.Values ?? ImmutableHashSet<Bot>.Empty;
+
+				return bots.Where(static bot => bot.IsConnectedAndLoggedOn).Select(static bot => new Measurement<int>(bot.SteamFriends.GetFriendCount(), new KeyValuePair<string, object?>(TagNames.BotName, bot.BotName)));
```

Comment:
I don’t like “one time series per bot” since it can and will blow up if you have a lot of accounts
Also bot name is not a consistent identifier (you can remove the bot named A and then add completely different account with the name A, all without restarting ASF).

Reply:
And steam id is not readable if you e.g. filter on a dashboard using some dropdown menu. That's why `TagNames.BotName` has the value `"bot"` and not `"account"`.

Of course I can add the steam-id as an additional tag. In some cases it might make sense.